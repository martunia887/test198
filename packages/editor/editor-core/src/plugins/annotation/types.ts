import * as React from 'react';
import { Node as PMNode } from 'prosemirror-model';

/*
  Inline Comments Component public API
*/
export type InlineCommentComponentProps = {
  /* annotation id (provided when viewing comment) */
  markerRef?: string;

  /* selected text (provided when creating comment) */
  textSelection?: string;

  /* DOM element having annotation (for positioning) */
  dom?: HTMLElement;

  /* callbacks */
  onSuccess: (id: string) => void;
  onDelete: (id: string) => void;

  /*
    Since the Editor loses focus when the actual comment is being made,
    we need to add a temporary mark in ProseMirror to show the selection 
    to the user. If the user dismisses the comment creation, the query mark 
    must be removed in the Editor using this callback
  */
  onCancel: () => void;
};

/*
  Reactions Component public API
*/
export type ReactionComponentProps = {
  // Coming later!
};

export interface AnnotationProvider {
  /* 
    Products can choose to have their own component for handling inline comments (and in a future release, reactions).
    We provide ExampleInlineCommentsComponent in the examples as reference implementation.
  */
  inlineCommentComponent?: React.ComponentType<InlineCommentComponentProps>;
  reactionComponent?: React.ComponentType<ReactionComponentProps>;
}

// actions
export enum AnnotationToolbarAction {
  SELECTION_CHANGE = 'SELECTION_CHANGE',
}

export enum AnnotationToolbarStateName {
  EDIT_ANNOTATION_TOOLBAR = 'EDIT_ANNOTATION_TOOLBAR',
}

export type EditAnnotationState = {
  type: AnnotationToolbarStateName.EDIT_ANNOTATION_TOOLBAR;
  pos: number;
  node: PMNode;
};

export type AnnotationToolbarState = EditAnnotationState | undefined;
