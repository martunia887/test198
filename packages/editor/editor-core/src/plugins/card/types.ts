import { INPUT_METHOD } from '../analytics';

export type CardAppearance = 'inline' | 'block';
export type CardType = 'smart-card' | 'custom' | 'unsupported';

export interface CardProvider {
  resolve(url: string, appearance: CardAppearance): Promise<any>;
}

export interface CardOptions {
  provider?: Promise<CardProvider>;
  resolveBeforeMacros?: string[];
}

export type Request = {
  pos: number;
  url: string;
  appearance: CardAppearance;
  compareLinkText: boolean;
  source: CardReplacementInputMethod;
};

export type CardPluginState = {
  requests: Request[];
  provider: CardProvider | null;
};

// actions
export type SetProvider = {
  type: 'SET_PROVIDER';
  provider: CardProvider | null;
};

export type Queue = {
  type: 'QUEUE';
  requests: Request[];
};

export type Resolve = {
  type: 'RESOLVE';
  url: string;
};

export type CardPluginAction = SetProvider | Queue | Resolve;

export type CardReplacementInputMethod =
  | INPUT_METHOD.CLIPBOARD
  | INPUT_METHOD.AUTO_DETECT
  | INPUT_METHOD.FORMATTING
  | INPUT_METHOD.MANUAL
  | INPUT_METHOD.TYPEAHEAD;
