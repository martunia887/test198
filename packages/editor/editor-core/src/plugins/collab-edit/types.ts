import { ReactElement } from 'react';
import { CollabEditProvider } from './provider';

export {
  Participant,
  InitData,
  RemoteData,
  ConnectionData,
  PresenceData,
  TelepointerData,
  SendableSelection,
} from '@atlaskit/editor-common';

export type InviteToEditComponentProps = {
  children: ReactElement<InviteToEditButtonProps>;
};

export type InviteToEditButtonProps = {
  onClick: (event: React.MouseEvent<HTMLElement>) => void;
  selected: boolean;
};

export interface CollabInviteToEditProps {
  inviteToEditHandler?: (event: React.MouseEvent<HTMLElement>) => void;
  isInviteToEditButtonSelected?: boolean;
  inviteToEditComponent?: React.ComponentType<InviteToEditComponentProps>;
}

export type CollabEditOptions = {
  provider?: Promise<CollabEditProvider>;
  userId?: string;
  useNativePlugin?: boolean;
  allowUnsupportedContent?: boolean;
} & CollabInviteToEditProps;
