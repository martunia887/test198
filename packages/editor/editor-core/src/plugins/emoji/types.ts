import {
  EmojiProvider,
  EmojiDescription,
  EmojiSearchResult,
} from '@atlaskit/emoji';

export interface EmojiProviderHandler {
  result: (result: EmojiSearchResult) => void;
}

export interface EmojiPluginState {
  provider?: EmojiProvider;
  providerHandler?: EmojiProviderHandler;
  emojis?: Array<EmojiDescription>;
}

export type EmojiPluginAction =
  | {
      type: 'SET_PROVIDER';
      provider?: EmojiProvider;
      providerHandler?: EmojiProviderHandler;
    }
  | {
      type: 'SET_EMOJIS';
      emojis: Array<EmojiDescription>;
    };
