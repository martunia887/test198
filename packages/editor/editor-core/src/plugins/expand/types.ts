export interface ExpandPluginState {}
export type ExpandPluginAction = {};
