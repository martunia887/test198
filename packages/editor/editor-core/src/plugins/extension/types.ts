import { ExtensionLayout } from '@atlaskit/adf-schema';
import { UpdateExtension, ExtensionProvider } from '@atlaskit/editor-common';
import { NodeWithPos } from 'prosemirror-utils';

export type ExtensionState = {
  layout: ExtensionLayout;
  allowBreakout: boolean;
  stickToolbarToBottom: boolean;
  showEditButton: boolean;
  updateExtension?: UpdateExtension<object>;
  nodeWithPos?: NodeWithPos;
  element?: HTMLElement;
  extensionProvider?: ExtensionProvider;
};

export type ExtensionAction =
  | {
      type: 'SET_NODE_INFO';
      data: { element?: HTMLElement; nodeWithPos?: NodeWithPos };
    }
  | {
      type: 'UPDATE_STATE';
      data: Partial<ExtensionState>;
    };
