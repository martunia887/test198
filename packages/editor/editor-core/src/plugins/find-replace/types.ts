export type Match = {
  /* Start position */
  start: number;
  /* End position */
  end: number;
};
