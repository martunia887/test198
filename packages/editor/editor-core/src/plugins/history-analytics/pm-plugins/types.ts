import { StepMap, Step } from 'prosemirror-transform';
import { Selection } from 'prosemirror-state';

export const pmHistoryPluginKey = 'history$';
export const DEPTH_OVERFLOW = 20;

export interface PmHistoryItem {
  map: StepMap;
  mirrorOffset?: number;
  selection?: Selection;
  step: Step;
}

export interface PmHistoryLeaf {
  values: PmHistoryItem[];
  length: number;
  forEach: (
    handler: (item: PmHistoryItem) => boolean | void,
    from?: number,
    to?: number,
  ) => void;
}

export interface PmHistoryBranch {
  items: PmHistoryLeaf;
  eventCount: number;
}

export interface PmHistoryPluginState {
  done: PmHistoryBranch;
  undone: PmHistoryBranch;
  prevMap?: StepMap;
  prevTime: number;
}

export interface PmHistoryPluginConfig {
  depth: number;
  newGroupDelay: number;
}
