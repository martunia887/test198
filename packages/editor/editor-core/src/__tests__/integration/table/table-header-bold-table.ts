import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';

import {
  editable,
  getDocFromElement,
  fullpage,
  quickInsert,
} from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';
import { TableCssClassName } from '../../../plugins/table/types';

BrowserTestCase(
  'should add strong mark on table header by default',
  { skip: [] },
  async client => {
    const CELL = `tr:first-child .${
      TableCssClassName.TABLE_HEADER_NODE_WRAPPER
    }:first-child .${TableCssClassName.CELL_NODEVIEW_WRAPPER} p`;
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        advanced: true,
      },
    });

    await page.click(editable);
    await quickInsert(page, 'Table');
    await page.waitForSelector(CELL);

    await page.click(CELL);
    await page.type(editable, 'this text should be bold');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);

BrowserTestCase(
  'should add strong mark on table header columns by default',
  { skip: [] },
  async client => {
    const CELL_NOT_BOLD = `tr:nth-child(2) .${
      TableCssClassName.TABLE_CELL_NODE_WRAPPER
    }:first-child .${TableCssClassName.CELL_NODEVIEW_WRAPPER} p`;
    const CELL_BOLD = `tr:nth-child(3) .${
      TableCssClassName.TABLE_HEADER_NODE_WRAPPER
    }:first-child .${TableCssClassName.CELL_NODEVIEW_WRAPPER} p`;
    const page = await goToEditorTestingExample(client);
    await mountEditor(page, {
      appearance: fullpage.appearance,
      allowTables: {
        advanced: true,
      },
    });

    await page.click(editable);
    await quickInsert(page, 'Table');

    await page.click(CELL_NOT_BOLD);
    await page.type(editable, 'this text should not be bold');

    await page.click(`[aria-label="Table options"]`);
    await page.click(`[aria-label="Header column"]`);

    await page.click(CELL_BOLD);
    await page.type(editable, 'this text should be bold');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchDocSnapshot();
  },
);
