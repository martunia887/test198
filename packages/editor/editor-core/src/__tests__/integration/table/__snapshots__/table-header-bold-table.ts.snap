// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add strong mark on table header by default 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "marks": Array [
                        Object {
                          "type": "strong",
                        },
                      ],
                      "text": "this text should be bold",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;

exports[`should add strong mark on table header columns by default 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "__autoSize": false,
        "isNumberColumnEnabled": false,
        "layout": "default",
      },
      "content": Array [
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "text": "this text should not be bold",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [
                  "strong",
                ],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "marks": Array [
                        Object {
                          "type": "strong",
                        },
                      ],
                      "text": "this text should be bold",
                      "type": "text",
                    },
                  ],
                  "type": "paragraph",
                },
              ],
              "type": "tableHeader",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
            Object {
              "attrs": Object {
                "background": null,
                "colspan": 1,
                "colwidth": null,
                "defaultMarks": Array [],
                "rowspan": 1,
              },
              "content": Array [
                Object {
                  "type": "paragraph",
                },
              ],
              "type": "tableCell",
            },
          ],
          "type": "tableRow",
        },
      ],
      "type": "table",
    },
  ],
  "type": "doc",
}
`;
